web/src/lib/utils/click-outside.ts

```diff
@@ -43,12 +43,12 @@ export function clickOutside(node: HTMLElement, options: Options = {}): ActionRe
   };
 
   document.addEventListener('click', handleClick, true);
-  document.addEventListener('keydown', handleKey, true);
+  node.addEventListener('keydown', handleKey, false);
 
   return {
     destroy() {
       document.removeEventListener('click', handleClick, true);
-      document.removeEventListener('keydown', handleKey, true);
+      node.removeEventListener('keydown', handleKey, false);
```

Comment:
Setting the listener on the document was causing both modals to close at once in scenarios where 2 overlapping modals are open at once. Setting the listener on the element itself and letting the event bubble up in the DOM solves this problem.

This does mean that modals now need `tabindex="-1"` on the body - if a mouse user is navigating the modal and presses escape without a form element focus, this guarantees that the modal will still close.